trace/kafka/kafka.go

```diff
@@ -74,18 +79,18 @@ func NewAsyncProducer(brokers []string, oo ...OptionFunc) (*AsyncProducer, error
 	cfg := sarama.NewConfig()
 	cfg.Version = sarama.V0_11_0_0
 
-	ap := AsyncProducer{cfg: cfg, chErr: make(chan error), tag: opentracing.Tag{Key: "type", Value: "async"}}
+	ap := AsyncProducer{cfg: cfg, chErr: make(chan error), tag: opentracing.Tag{Key: "type", Value: "async"}, enc: json.Encode}
 
 	for _, o := range oo {
 		err := o(&ap)
 		if err != nil {
-			return nil, err
+			return nil, errors.Errorf("Could not apply OptionFunc '%v' to producer : %v", runtime.FuncForPC(reflect.ValueOf(o).Pointer()).Name(), err)
```

Comment:
Pls avoid reflection. Only use it if nothing else works... the error from the option will have an appropriate message.